packages/zoe/src/contracts/priceAggregatorChainlink.js

```diff
@@ -0,0 +1,743 @@
+    }
+
+    const error = validateOracleRound(_oracle, roundId);
+    if (error.length !== 0) {
+      eligibleToSubmit = false;
+    }
+
+    return {
+      eligibleForSpecificRound: eligibleToSubmit,
+      queriedRoundId: roundId,
+      oracleStatus: oracle.latestSubmission,
+      startedAt,
+      roundTimeout,
+      oracleCount: oracleCount(),
+    };
+  };
+
+  /**
+   * @param {Instance} _oracle
+   * @param {bigint} _queriedRoundId
+   * @returns {boolean}
+   */
+  const eligibleForSpecificRound = (_oracle, _queriedRoundId) => {
+    const error = validateOracleRound(_oracle, _queriedRoundId);
+    if (rounds.get(_queriedRoundId).startedAt > 0) {
+      return acceptingSubmissions(_queriedRoundId) && error.length === 0;
+    } else {
+      return delayed(_oracle, _queriedRoundId) && error.length === 0;
+    }
+  };
+
+  /**
+   * @param {Instance} _oracle
+   * @returns {bigint}
+   */
+  const getStartingRound = _oracle => {
+    const currentRound = reportingRoundId;
+    if (
+      currentRound !== BigInt(0) &&
+      currentRound === oracleStatuses.get(_oracle).endingRound
+    ) {
+      return currentRound;
+    }
+    return add(currentRound, 1);
+  };
+
+  /** @type {PriceAggregatorCreatorFacet} */
+  const creatorFacet = Far('PriceAggregatorChainlinkCreatorFacet', {
+    async initializeQuoteMint(quoteMint) {
+      const quoteIssuerRecord = await zcf.saveIssuer(
+        E(quoteMint).getIssuer(),
+        'Quote',
+      );
+      quoteKit = {
+        ...quoteIssuerRecord,
+        mint: quoteMint,
+      };
+
+      const paKit = makeOnewayPriceAuthorityKit({
+        createQuote: makeCreateQuote(),
+        notifier,
+        quoteIssuer: quoteKit.issuer,
+        timer,
+        actualBrandIn: brandIn,
+        actualBrandOut: brandOut,
+      });
+      ({ priceAuthority } = paKit);
+    },
+
+    // unlike the median case, no query argument is passed, since polling behavior is undesired
+    async initOracle(oracleInstance) {
+      assert(quoteKit, X`Must initializeQuoteMint before adding an oracle`);
+
+      /** @type {OracleRecord} */
+      const record = { querier: undefined, lastSample: 0 };
+
+      /** @type {Set<OracleRecord>} */
+      let records;
+      if (instanceToRecords.has(oracleInstance)) {
+        records = instanceToRecords.get(oracleInstance);
+      } else {
+        records = new Set();
+        instanceToRecords.init(oracleInstance, records);
+
+        const oracleStatus = makeOracleStatus(
+          /* startingRound = */ getStartingRound(oracleInstance),
+          /* endingRound = */ ROUND_MAX,
+          /* lastReportedRound = */ BigInt(0),
+          /* lastStartedRound = */ BigInt(0),
+          /* latestSubmission = */ BigInt(0),
+          /* index = */ oracleStatuses.keys().length,
+        );
+        oracleStatuses.init(oracleInstance, oracleStatus);
+      }
+      records.add(record);
+
+      const pushResult = async (_roundIdRaw, _submissionRaw) => {
```

Comment:
FWIW, the oracle protocol can only call pushResult with one argument.  Would it make sense to do:

```suggestion
      const pushResult = async ({ roundId: _roundIdRaw = undefined, data: _submissionRaw }) => {
```

so that callers can do: `pushResult({ roundId: 3n, data: '19919' })`?

Then, if `roundId` is not specified in the input, you could default to the suggested round for this oracleInstance?

Reply:
Oh interesting, yeah, that seems clean